cpp/find_latest_journal_year.cc

```diff
@@ -157,12 +157,28 @@ void ProcessRecords(MARC::Reader * const marc_reader, std::unordered_map<std::st
             continue;
 
         std::string year_as_string;
-        for (const auto &field : record.getTagRange("936")) {
-            const MARC::Subfields subfields(field.getSubfields());
-            year_as_string = subfields.getFirstSubfieldWithCode('j');
-            if (not year_as_string.empty())
-                break;
+
+        std::vector<std::string> subfields;
+        std::vector<std::string> filtered_dates;
+        for (const auto &field : record.getTagRange("773")) {
+            if (field.getIndicator1() == '1') {
+                for (const auto &subfield : field.getSubfields()) {
```

Comment:
- Instead of iterating all subfields, you should only iterate subfields with the code `g`.
- You should only call `emplace_back` if the key before the `:` equals `year`.